ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -518,14 +544,25 @@ private byte[] serializeValue(
   private static FormatInfo addSerializerMissingFormatFields(
       final FormatInfo formatInfo,
       final String topicName,
-      final boolean isKey
+      final boolean isKey,
+      final Optional<Integer> schemaId
   ) {
     // Just add missing fields required for serialization SR formats
     final Format format = FormatFactory.fromName(formatInfo.getFormat());
     if (!format.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
       return formatInfo;
     }
 
+    if (schemaId.isPresent()) {
+      final ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+      propertiesBuilder.putAll(formatInfo.getProperties());
+      if (!formatInfo.getProperties().containsKey(ConnectProperties.SCHEMA_ID)) {
+        propertiesBuilder.put(ConnectProperties.SCHEMA_ID, String.valueOf(schemaId.get()));
```

Comment:
Do you need the `String.valueOf`? I thought integers would work fine.

Reply:
The IDE was complaining